src/audio/copier/copier.c

```diff
@@ -228,11 +230,23 @@ static int create_host(struct comp_dev *parent_dev, struct copier_data *cd,
 	ipc_host.direction = dir;
 	ipc_host.dma_buffer_size = copier_cfg->gtw_cfg.dma_buffer_size;
 
-	dev = drv->ops.create(drv, config, &ipc_host);
-	if (!dev) {
-		ret = -EINVAL;
+	dev = comp_alloc(drv, sizeof(*dev));
+	if (!dev)
```

Comment:
you need to set the return value here . ret = -ENOMEM and then goto e_bug please

Reply:
ok